src/App.tsx

```diff
@@ -15,36 +16,105 @@
   'Garlic',
 ];
 
+enum SortType {
+  alphabet = 'alphabet',
+  length = 'length',
+  default = '',
+}
+
+interface SortParams {
+  sortBy: SortType;
+  isReversed: boolean;
+}
+
+function getPreparedGoods(goods: string[], { sortBy, isReversed }: SortParams) {
+  const preparedGoods = [...goods];
+
+  if (sortBy !== SortType.default) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case SortType.length:
+          return good1.length - good2.length;
+
+        case SortType.alphabet:
+          return good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState<SortType>(SortType.default);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortBy,
+    isReversed,
+  });
+
+  const handleResetButton = () => {
+    setSortBy(SortType.default);
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortBy !== SortType.alphabet,
+          })}
+          onClick={() => setSortBy(SortType.alphabet)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortBy !== SortType.length,
+          })}
+          onClick={() => setSortBy(SortType.length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setIsReversed(!isReversed)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortBy !== SortType.default || isReversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={handleResetButton}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {visibleGoods.map(good => (
+          <li key={good} data-cy="Good">
+            {good}
+          </li>
+        ))}
```

Comment:
[CODE KNOWLEDGE] - When rendering a list, make sure to add a unique `key` prop. Using the good's name as a key is risky if the list can contain duplicates. Consider using an index or a unique identifier.